### PR TITLE
use TokenCredential in services

### PR DESCRIPTION
**Pull request by ctaggart**
Please review for #53. I made these changes manually to update `cargo run --example group_create` so that it works with `TokenCredential`. Once we figure out the design, I'll update the AutoRust code gen.

### CODE REVIEW

File: sdk/core/Cargo.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ async-trait = "0.1.36"
 oauth2 = { version = "4.0.0-alpha.2" }
 
 [dev-dependencies]
-tokio = "0.3"
+tokio = "0.2"
```
Comment (ctaggart) [inline sdk/core/Cargo.toml:37]: I got some runtime errors regarding the hyper dependency. I think we may need to wait for hyper to upgrade.

File: sdk/core/src/lib.rs [hunk 2]
```diff
@@ -23,7 +23,7 @@ pub mod range;
 mod stored_access_policy;
 pub mod util;
 
-use errors::AzureError;
+pub use errors::AzureError;
```
Comment (ctaggart) [inline sdk/core/src/lib.rs:26]: Any reason why it wasn't pub?

File: services/mgmt/resources/src/lib.rs [hunk 3]
```diff
@@ -175,12 +175,14 @@ pub struct OperationConfig {
     pub api_version: String,
     pub client: reqwest::Client,
     pub base_path: String,
-    pub bearer_access_token: Option<String>,
+    pub bearer_access_token: Option<String>, // TODO remove
+    pub token_credential: Option<Box<dyn azure_core::TokenCredential>>,
+    pub token_credential_scope: String,
```
Comment (ctaggart) [inline services/mgmt/resources/src/lib.rs:180]: Shouldn't it be a list of scopes?
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-acquire-cache-tokens#scopes-when-acquiring-tokens
For v2.0, it is still in the `scope` http query param, but it is a space separated list. Let's looks and see what other SDKs do.

File: sdk/core/src/lib.rs (hunk 2 above)
Comment (rylev) [reply sdk/core/src/lib.rs:26]: Not that I know of

File: sdk/core/Cargo.toml (hunk 1 above)
Comment (ctaggart) [reply sdk/core/Cargo.toml:37]: - hyper v0.14 https://github.com/hyperium/hyper/issues/2302
- reqwest ? https://github.com/seanmonstar/reqwest/issues/1060

Comment (ctaggart) [reply sdk/core/Cargo.toml:37]: I created #60 to track upgrading.

File: services/mgmt/resources/src/lib.rs (hunk 3 above)
Comment (ctaggart) [reply services/mgmt/resources/src/lib.rs:180]: I added an issue to review this in the not too distant future: https://github.com/Azure/azure-sdk-for-rust/issues/61

File: sdk/core/src/lib.rs (hunk 2 above)
Comment (ctaggart) [reply sdk/core/src/lib.rs:26]: It is already exposed as `azure_core::errors::AzureError`.

**Comment by ctaggart** [comment]
It may be easiest to review the AutoRust changes for this in https://github.com/ctaggart/autorust/pull/69.

**Comment by ctaggart** [comment]
@rylev, can I get an approval of this one?

**Review by rylev — APPROVED** [review]

**Pull request merged**